pyhive/sqlalchemy_hive.py

```diff
@@ -517,6 +518,37 @@ def _check_unicode_description(self, connection):
         # We decode everything as UTF-8
         return True
 
+
+from sqlalchemy.ext.compiler import compiles
+from sqlalchemy.sql.expression import Select
+
+
+@compiles(Select)
+def select_star_if_all_columns(select, compiler, **kwargs):
+    froms = select.froms
+    select_columns = list(select.columns.keys())
+    from_columns = list(chain.from_iterable((f.columns.keys() for f in froms)))
```

Comment:
Would a cleaner way be to whitelist queries where the following conditions are true:

* the `from` clause is a `Table` instance
* the query has `limit` clause
* all the columns from the `from` clause are in the query
?